inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,260 @@
+    """
+    if doc_type in DOCUMENT_TYPE_MAP:
+        return DOCUMENT_TYPE_MAP[doc_type]
+    # Theses need special treatment, because bibtex differentiates between their types:
+    elif doc_type == 'thesis' and obj['thesis_info']['degree_type'] == 'phd':
+        return 'phdthesis'
+    elif doc_type == 'thesis' and obj['thesis_info']['degree_type'] == 'master':
+        return 'mastersthesis'
+    # Other types of theses (other, diploma, bachelor, laurea, habilitation) don't have separate types in bibtex:
+    elif doc_type == 'thesis':
+        return 'misc'  # TODO: maybe use mastersthesis with a type specifying the degree?
+    # Report numbers imply report type
+    # elif doc_type == 'note' and 'report_numbers' in obj:
+    #     return 'techreport'
+    else:
+        return 'misc'
+
+
+def bibtex_type_and_fields(data):
+    """Returns a bibtex document type and fields for an entry."""
+    bibtex_doc_types = []
+    for idx, doc_type in enumerate(data['document_type']):
+        bibtex_doc_types.append(bibtex_document_type(doc_type, data))
+
+    # Preference towards article, as it's more prestigious to have sth published
+    if 'article' in bibtex_doc_types:
+        return 'article', FIELDS_FOR_ENTRY_TYPE['article']
+
+    bibtex_doc_type = bibtex_doc_types[0] if bibtex_doc_types else 'misc'
+    return bibtex_doc_type, FIELDS_FOR_ENTRY_TYPE[bibtex_doc_type]
+
+
+extractor = make_extractor()
+
+
+def get_authors_with_role(authors, role):
+    """
+    Extract names of people from an authors field given their roles.
+    :param authors: Authors field of the record.
+    :param role: String specifying the role 'author', 'editor', etc.
+    :return: List of names of people.
+    """
+    filtered = []
+    for author in authors:
+        if 'inspire_roles' in author and role in author['inspire_roles']:
+            filtered.append(author['full_name'])
+        elif 'inspire_roles' not in author and role == 'author':
+            filtered.append(author['full_name'])
+    return filtered
+
+
+def eprint_new_style(eprint):
+    """
+    Given an arXiv eprint sting detect new (1234.5678) vs old style (hep-ph/123456)
+    :param eprint: Eprint string.
+    :return: True if new style, else False.
+    """
+    return '/' not in eprint and ' ' not in eprint
+
+
+def get_publication_info(data):
+    """
+    Get publication_info field from data.
+    :param data: Pre-loaded record.
+    :return: Return publication_info field if exists, else empty dictionary.
+    """
+    return data.get('publication_info', {})
+
+# Functions below describe where the non-obvious data is located
+
+
+@extractor('author')
+def get_author(data, doc_type):
+    """
+    Note: Only used to generate author field if corporate_author is the author.
+    """
+    if 'corporate_author' in data:
+        return ', '.join(data['corporate_author'])
+
+
+@extractor('journal')
+def get_journal(data, doc_type):
+    return get_publication_info(data).get('journal')
```

Comment:
you could use [`get_value`](https://github.com/inspirehep/inspire-utils/blob/master/inspire_utils/record.py#L31) instead of nested `get`s.